godel_process_planning/src/godel_process_planning.cpp

```diff
@@ -0,0 +1,46 @@
+#include "godel_process_planning/godel_process_planning.h"
+#include <moveit/robot_model_loader/robot_model_loader.h>
+
+godel_process_planning::ProcessPlanningManager::ProcessPlanningManager(const std::string &world_frame,
+                                                                       const std::string &blend_group,
+                                                                       const std::string &blend_tcp,
+                                                                       const std::string &keyence_group,
+                                                                       const std::string &keyence_tcp,
+                                                                       const std::string &robot_model_plugin)
+  : plugin_loader_("descartes_core", "descartes_core::RobotModel")
+  , blend_group_name_(blend_group)
+  , keyence_group_name_(keyence_group)
+{
+  // Attempt to load and initialize the blending robot model
+  blend_model_ = plugin_loader_.createInstance(robot_model_plugin);
+  if (!blend_model_)
+  {
+    throw std::runtime_error(std::string("Could not load: ") + robot_model_plugin);
+  }
+
+  if (!blend_model_->initialize("robot_description", blend_group, world_frame, blend_tcp))
+  {
+    throw std::runtime_error("Unable to initialize blending robot model");
+  }
+
+  // Attempt to load and initialize the scanning/keyence robot model
+  keyence_model_ = plugin_loader_.createInstance(robot_model_plugin);
+  if (!keyence_model_)
+  {
+    throw std::runtime_error(std::string("Could not load: ") + robot_model_plugin);
+  }
+
+  if (!keyence_model_->initialize("robot_description", keyence_group, world_frame, keyence_tcp))
```

Comment:
We should probably modify  the descartes RobotModel initialize() method so that it takes a moveit RobotModel instead of a string.  This should prevent having to load the robot description rom the ros parameter multiple times.  Loading large complex models can take a lot of time.